components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/utils/AdminAttributeUtilTest.java

```diff
@@ -0,0 +1,191 @@
+    public void testUpdateAdminUser(boolean validateSCIMID) throws Exception {
+        String adminUsername = "admin";
+
+        mockStatic(SCIMCommonComponentHolder.class);
+        mockStatic(ClaimsMgtUtil.class);
+        mockStatic(IdentityTenantUtil.class);
+        when(SCIMCommonComponentHolder.getRealmService()).thenReturn(realmService);
+        when(realmService.getTenantUserRealm(anyInt())).thenReturn(userRealm);
+        when(userRealm.getUserStoreManager()).thenReturn(userStoreManager);
+        when(userStoreManager.isSCIMEnabled()).thenReturn(true);
+        when(ClaimsMgtUtil.getAdminUserNameFromTenantId(eq(realmService), anyInt())).thenReturn(adminUsername);
+        when(IdentityTenantUtil.getRealmService()).thenReturn(realmService);
+        when(userStoreManager.getUserClaimValue(anyString(), anyString(), anyString())).thenReturn("");
+
+        ArgumentCaptor<Map> argument = ArgumentCaptor.forClass(Map.class);
+        adminAttributeUtil.updateAdminUser(-1234, validateSCIMID);
+        verify(userStoreManager).setUserClaimValues(anyString(), argument.capture(), anyString());
+
+        assertEquals(adminUsername, argument.getValue().get(SCIMConstants.UserSchemaConstants.USER_NAME_URI));
+    }
+
+    @Test(expectedExceptions = UserStoreException.class)
+    public void testUpdateAdminUser1() throws Exception {
+        mockStatic(SCIMCommonComponentHolder.class);
+        mockStatic(ClaimsMgtUtil.class);
+        mockStatic(IdentityTenantUtil.class);
+        when(SCIMCommonComponentHolder.getRealmService()).thenReturn(realmService);
+        when(realmService.getTenantUserRealm(anyInt())).thenReturn(userRealm);
+        when(userRealm.getUserStoreManager()).thenReturn(userStoreManager);
+        when(userStoreManager.isSCIMEnabled()).thenThrow(new UserStoreException());
+
+        adminAttributeUtil.updateAdminUser(-1234, true);
+        verify(userStoreManager.isSCIMEnabled());
+    }
+
+    @DataProvider(name = "testUpdateAdminGroupData")
+    public Object[][] testUpdateAdminGroupData() {
+        return new Object[][]{
+                {"carbon.super"},
+                {null}
+        };
+    }
+
+    @Test(dataProvider = "testUpdateAdminGroupData")
+    public void testUpdateAdminGroup(String domainName) throws Exception {
+        String roleNameWithDomain = "TESTDOMAIN/admin";
+
+        mockStatic(SCIMCommonComponentHolder.class);
+        mockStatic(ClaimsMgtUtil.class);
+        mockStatic(IdentityTenantUtil.class);
+        mockStatic(UserCoreUtil.class);
+        mockStatic(IdentityUtil.class);
+        mockStatic(SCIMCommonUtils.class);
+        when(SCIMCommonComponentHolder.getRealmService()).thenReturn(realmService);
+        when(realmService.getTenantUserRealm(anyInt())).thenReturn(userRealm);
+        when(userRealm.getUserStoreManager()).thenReturn(userStoreManager);
+        when(userStoreManager.isSCIMEnabled()).thenReturn(true);
+        when(userStoreManager.getTenantId()).thenReturn(-1234);
```

Comment:
Better to use MultitenantConstants.SUPER_TENANT_ID

Reply:
Updated the PR.